### PR TITLE
888-UseTransformHeadersFromTheHttpSharedLib

### PR DESCRIPTION
**Pull request by gibaros**
Issue#957-UseTransformHeadersFromSharedLib

- update dependencies
- add dependency check and update scripts
- remove NOTIFICATION_TOPIC_TEMPLATE
- use additional shared lib enums
- remove lib kafka produce.js
- remove lib utility.js
- remove handlers notification callback.js and replace its sendCallback with sharelib sendRequest
- remove infile enums and replace with sharelib Enums

### CODE REVIEW

File: src/domain/bulkTransfer/transformer.js [hunk 1]
```diff
@@ -57,14 +57,14 @@ const transformHeaders = (headers, config) => {
   var normalizedHeaders = {}
 
   // check to see if FSPIOP-Destination header has been left out of the initial request. If so then add it.
-  if (!normalizedKeys.hasOwnProperty(ENUM.headers.FSPIOP.DESTINATION)) {
-    headers[ENUM.headers.FSPIOP.DESTINATION] = ''
+  if (!normalizedKeys.hasOwnProperty(HttpEnum.headers.FSPIOP.DESTINATION)) {
+    headers[HttpEnum.headers.FSPIOP.DESTINATION] = ''
```
Comment (rmothilal) [inline src/domain/bulkTransfer/transformer.js]: Needs to be uppercase.Headers

File: src/domain/bulkTransfer/transformer.js [hunk 2]
```diff
@@ -57,14 +57,14 @@ const transformHeaders = (headers, config) => {
...
   }
 
   for (var headerKey in headers) {
     var headerValue = headers[headerKey]
     switch (headerKey.toLowerCase()) {
-      case (ENUM.headers.GENERAL.DATE):
+      case (HttpEnum.headers.GENERAL.DATE):
```
Comment (rmothilal) [inline src/domain/bulkTransfer/transformer.js]: Needs to be uppercase.Headers

File: src/domain/bulkTransfer/transformer.js [hunk 3]
```diff
@@ -80,21 +80,21 @@ const transformHeaders = (headers, config) => {
         }
         normalizedHeaders[headerKey] = tempDate
         break
-      case (ENUM.headers.GENERAL.CONTENT_LENGTH):
+      case (HttpEnum.headers.GENERAL.CONTENT_LENGTH):
```
Comment (rmothilal) [inline src/domain/bulkTransfer/transformer.js]: Needs to be uppercase.Headers

File: src/domain/bulkTransfer/transformer.js [hunk 4]
```diff
@@ -80,21 +80,21 @@ const transformHeaders = (headers, config) => {
...
         // Do nothing here, do not map. This will be inserted correctly by the Hapi framework.
         break
-      case (ENUM.headers.FSPIOP.URI):
+      case (HttpEnum.headers.FSPIOP.URI):
```
Comment (rmothilal) [inline src/domain/bulkTransfer/transformer.js]: Needs to be uppercase.Headers

File: src/domain/bulkTransfer/transformer.js [hunk 5]
```diff
@@ -80,21 +80,21 @@ const transformHeaders = (headers, config) => {
...
         // Check to see if we find a regex match the source header containing the switch name.
         // If so we include the uri otherwise we remove it.
 
-        if (headers[normalizedKeys[ENUM.headers.FSPIOP.SOURCE]].match(ENUM.headers.FSPIOP.SWITCH.regex) === null) {
+        if (headers[normalizedKeys[HttpEnum.headers.FSPIOP.SOURCE]].match(HttpEnum.headers.FSPIOP.SWITCH.regex) === null) {
```
Comment (rmothilal) [inline src/domain/bulkTransfer/transformer.js]: Needs to be uppercase.Headers

File: src/domain/bulkTransfer/transformer.js [hunk 6]
```diff
@@ -80,21 +80,21 @@ const transformHeaders = (headers, config) => {
...
           normalizedHeaders[headerKey] = headerValue
         }
         break
-      case (ENUM.headers.FSPIOP.HTTP_METHOD):
+      case (HttpEnum.headers.FSPIOP.HTTP_METHOD):
```
Comment (rmothilal) [inline src/domain/bulkTransfer/transformer.js]: Needs to be uppercase.Headers

File: src/domain/bulkTransfer/transformer.js [hunk 7]
```diff
@@ -80,21 +80,21 @@ const transformHeaders = (headers, config) => {
...
         // Check to see if we find a regex match the source header containing the switch name.
         // If so we include the signature otherwise we remove it.
-        if (headers[normalizedKeys[ENUM.headers.FSPIOP.SOURCE]].match(ENUM.headers.FSPIOP.SWITCH.regex) === null) {
+        if (headers[normalizedKeys[HttpEnum.headers.FSPIOP.SOURCE]].match(HttpEnum.headers.FSPIOP.SWITCH.regex) === null) {
```
Comment (rmothilal) [inline src/domain/bulkTransfer/transformer.js]: Needs to be uppercase.Headers

File: src/domain/bulkTransfer/transformer.js [hunk 8]
```diff
@@ -112,27 +112,27 @@ const transformHeaders = (headers, config) => {
           }
         }
         break
-      case (ENUM.headers.FSPIOP.SOURCE):
+      case (HttpEnum.headers.FSPIOP.SOURCE):
```
Comment (rmothilal) [inline src/domain/bulkTransfer/transformer.js]: Needs to be uppercase.Headers

File: src/domain/bulkTransfer/transformer.js [hunk 9]
```diff
@@ -112,27 +112,27 @@ const transformHeaders = (headers, config) => {
...
         normalizedHeaders[headerKey] = config.sourceFsp
         break
-      case (ENUM.headers.FSPIOP.DESTINATION):
+      case (HttpEnum.headers.FSPIOP.DESTINATION):
```
Comment (rmothilal) [inline src/domain/bulkTransfer/transformer.js]: Needs to be uppercase.Headers

File: src/domain/bulkTransfer/transformer.js [hunk 10]
```diff
@@ -112,27 +112,27 @@ const transformHeaders = (headers, config) => {
...
         normalizedHeaders[headerKey] = config.destinationFsp
         break
       default:
         normalizedHeaders[headerKey] = headerValue
     }
   }
 
-  if (normalizedHeaders[normalizedKeys[ENUM.headers.FSPIOP.SOURCE]].match(ENUM.headers.FSPIOP.SWITCH.regex) !== null) {
+  if (normalizedHeaders[normalizedKeys[HttpEnum.headers.FSPIOP.SOURCE]].match(HttpEnum.headers.FSPIOP.SWITCH.regex) !== null) {
```
Comment (rmothilal) [inline src/domain/bulkTransfer/transformer.js]: Needs to be uppercase.Headers

File: src/domain/bulkTransfer/transformer.js [hunk 11]
```diff
@@ -112,27 +112,27 @@ const transformHeaders = (headers, config) => {
...
     // Check to see if we find a regex match the source header containing the switch name.
     // If so we remove the signature added by default.
-    delete normalizedHeaders[ENUM.headers.FSPIOP.SIGNATURE]
+    delete normalizedHeaders[HttpEnum.headers.FSPIOP.SIGNATURE]
```
Comment (rmothilal) [inline src/domain/bulkTransfer/transformer.js]: Needs to be uppercase.Headers

File: src/domain/bulkTransfer/transformer.js [hunk 12]
```diff
@@ -112,27 +112,27 @@ const transformHeaders = (headers, config) => {
...
     // Aslo remove FSPIOP-URI and make FSPIOP-HTTP-Method ALL-CAPS #737
-    delete normalizedHeaders[ENUM.headers.FSPIOP.URI]
+    delete normalizedHeaders[HttpEnum.headers.FSPIOP.URI]
```
Comment (rmothilal) [inline src/domain/bulkTransfer/transformer.js]: Needs to be uppercase.Headers

File: src/domain/bulkTransfer/transformer.js [hunk 13]
```diff
@@ -112,27 +112,27 @@ const transformHeaders = (headers, config) => {
...
   }
 
-  if (config && config.httpMethod !== ENUM.methods.FSPIOP_CALLBACK_URL_TRANSFER_POST) {
-    delete normalizedHeaders[ENUM.headers.GENERAL.ACCEPT]
+  if (config && config.httpMethod !== HttpEnum.methods.FSPIOP_CALLBACK_URL_TRANSFER_POST) {
+    delete normalizedHeaders[HttpEnum.headers.GENERAL.ACCEPT]
```
Comment (rmothilal) [inline src/domain/bulkTransfer/transformer.js]: Needs to be uppercase.Headers

File: src/handlers/notification/callbacks.js [hunk 14]
```diff
@@ -52,7 +51,7 @@ const Enum = require('../../lib/enum')
 const sendCallback = async (url, method, headers, message, cid, sourceFsp, destinationFsp) => {
   // validate incoming request parameters are not null or undefined
   if (!url || !method || !headers || !message || !cid || !sourceFsp || !destinationFsp) {
-    throw new Error(Enum.errorMessages.MISSINGFUNCTIONPARAMETERS)
+    throw new Error('Missing parameters for function')
```
Comment (rmothilal) [inline src/handlers/notification/callbacks.js]: Please switch to sendRequest from shared library

File: src/handlers/notification/index.js [hunk 15]
```diff
@@ -163,43 +160,43 @@ const processMessage = async (msg) => {
     let id = JSON.parse(decodedPayload.body.toString()).transferId || (content.uriParams && content.uriParams.id)
     let payloadForCallback = decodedPayload.body.toString()
 
-    if (actionLower === ENUM.transferEventAction.BULK_PREPARE && statusLower === ENUM.messageStatus.SUCCESS) {
```
Comment (rmothilal) [inline src/handlers/notification/index.js]: Use enum from events in shared library

File: src/handlers/notification/index.js [hunk 16]
```diff
@@ -163,43 +160,43 @@ const processMessage = async (msg) => {
...
+      let callbackURLTo = await Participant.getEndpoint(to, HttpEnum.RestMethods.POST, id)
+      let methodTo = HttpEnum.RestMethods.POST
       Logger.debug(`Notification::processMessage - Callback.sendCallback(${callbackURLTo}, ${methodTo}, ${JSON.stringify(content.headers)}, ${payloadForCallback}, ${id}, ${from}, ${to})`)
       let bulkResponseMessage = await BulkTransfer.getBulkTransferResultByMessageIdDestination(messageId, to)
       responsePayload.individualTransferResults = bulkResponseMessage.individualTransferResults
       return Callback.sendCallback(callbackURLTo, methodTo, content.headers, JSON.stringify(responsePayload), id, from, to)
     }
 
-    if (actionLower === ENUM.transferEventAction.BULK_PREPARE && statusLower !== ENUM.messageStatus.SUCCESS) {
+    if (actionLower === 'bulk-prepare' && statusLower !== 'success') {
```
Comment (rmothilal) [inline src/handlers/notification/index.js]: Use enum from events in shared library

File: src/handlers/notification/index.js [hunk 17]
```diff
@@ -163,43 +160,43 @@ const processMessage = async (msg) => {
...
-      let callbackURLTo = await Participant.getEndpoint(to, FSPIOP_CALLBACK_URL_BULK_TRANSFER_ERROR, id)
-      let methodFrom = ENUM.methods.FSPIOP_CALLBACK_URL_BULK_TRANSFER_ERROR
+      let callbackURLTo = await Participant.getEndpoint(to, HttpEnum.RestMethods.PUT, id)
+      let methodFrom = HttpEnum.RestMethods.PUT
       Logger.debug(`Notification::processMessage - Callback.sendCallback(${callbackURLTo}, ${methodFrom}, ${JSON.stringify(content.headers)}, ${payloadForCallback}, ${id}, ${from}, ${to})`)
       return Callback.sendCallback(callbackURLTo, methodFrom, content.headers, payloadForCallback, id, from, to)
     }
 
-    if (actionLower === ENUM.transferEventAction.BULK_COMMIT && statusLower === ENUM.messageStatus.SUCCESS) {
+    if (actionLower === 'bulk-commit' && statusLower === 'success') {
```
Comment (rmothilal) [inline src/handlers/notification/index.js]: Use enum from events in shared library

File: src/handlers/notification/index.js [hunk 18]
```diff
@@ -163,43 +160,43 @@ const processMessage = async (msg) => {
...
+      let callbackURLTo = await Participant.getEndpoint(to, HttpEnum.RestMethods.PUT, id)
+      let methodTo = HttpEnum.RestMethods.PUT
       Logger.debug(`Notification::processMessage - Callback.sendCallback(${callbackURLTo}, ${methodTo}, ${JSON.stringify(content.headers)}, ${JSON.stringify(responsePayload)}, ${id}, ${from}, ${to})`)
       let bulkResponseMessage = await BulkTransfer.getBulkTransferResultByMessageIdDestination(messageId, to)
       responsePayload.individualTransferResults = bulkResponseMessage.individualTransferResults
       return Callback.sendCallback(callbackURLTo, methodTo, content.headers, JSON.stringify(responsePayload), id, from, to)
     }
 
-    if (actionLower === ENUM.transferEventAction.BULK_COMMIT && statusLower !== ENUM.messageStatus.SUCCESS) {
+    if (actionLower === 'bulk-commit' && statusLower !== 'success') {
```
Comment (rmothilal) [inline src/handlers/notification/index.js]: Use enum from events in shared library

File: src/handlers/notification/callbacks.js (hunk 14 above)
Comment (gibaros) [reply src/handlers/notification/callbacks.js]: So, remove and replace:

const sendCallback = async (url, method, headers, message, cid, sourceFsp, destinationFsp) => {
  // validate incoming request parameters are not null or undefined
  if (!url || !method || !headers || !message || !cid || !sourceFsp || !destinationFsp) {
    throw new Error('Missing parameters for function')
  }

with sendRequest from shared lib?

File: src/domain/bulkTransfer/transformer.js (hunk 1 above)
Comment (gibaros) [reply src/domain/bulkTransfer/transformer.js]: Addressed

File: src/domain/bulkTransfer/transformer.js (hunk 2 above)
Comment (gibaros) [reply src/domain/bulkTransfer/transformer.js]: Addressed

File: src/domain/bulkTransfer/transformer.js (hunk 3 above)
Comment (gibaros) [reply src/domain/bulkTransfer/transformer.js]: Addressed

File: src/domain/bulkTransfer/transformer.js (hunk 4 above)
Comment (gibaros) [reply src/domain/bulkTransfer/transformer.js]: Addressed

File: src/domain/bulkTransfer/transformer.js (hunk 5 above)
Comment (gibaros) [reply src/domain/bulkTransfer/transformer.js]: Addressed

File: src/domain/bulkTransfer/transformer.js (hunk 6 above)
Comment (gibaros) [reply src/domain/bulkTransfer/transformer.js]: Addressed

File: src/domain/bulkTransfer/transformer.js (hunk 7 above)
Comment (gibaros) [reply src/domain/bulkTransfer/transformer.js]: Addressed

File: src/domain/bulkTransfer/transformer.js (hunk 8 above)
Comment (gibaros) [reply src/domain/bulkTransfer/transformer.js]: Addressed

File: src/domain/bulkTransfer/transformer.js (hunk 9 above)
Comment (gibaros) [reply src/domain/bulkTransfer/transformer.js]: Addressed

File: src/domain/bulkTransfer/transformer.js (hunk 10 above)
Comment (gibaros) [reply src/domain/bulkTransfer/transformer.js]: Addressed

File: src/domain/bulkTransfer/transformer.js (hunk 11 above)
Comment (gibaros) [reply src/domain/bulkTransfer/transformer.js]: Addressed

File: src/domain/bulkTransfer/transformer.js (hunk 12 above)
Comment (gibaros) [reply src/domain/bulkTransfer/transformer.js]: Addressed

File: src/domain/bulkTransfer/transformer.js (hunk 13 above)
Comment (gibaros) [reply src/domain/bulkTransfer/transformer.js]: Addressed

File: src/handlers/notification/callbacks.js (hunk 14 above)
Comment (gibaros) [reply src/handlers/notification/callbacks.js]: Addressed, file deleted in order to use the sendRequest from shared lib.

File: src/handlers/notification/index.js (hunk 15 above)
Comment (gibaros) [reply src/handlers/notification/index.js]: Addressed

File: src/handlers/notification/index.js (hunk 16 above)
Comment (gibaros) [reply src/handlers/notification/index.js]: Addressed

File: src/handlers/notification/index.js (hunk 17 above)
Comment (gibaros) [reply src/handlers/notification/index.js]: Addressed

File: src/handlers/notification/index.js (hunk 18 above)
Comment (gibaros) [reply src/handlers/notification/index.js]: Addressed

**Comment by gibaros** [comment]
PR ready for review, all changes requested by @rmothilal have been addressed. PR currently blocked as Unit Tests are not passing, but need to double check as they were not passing before PR was created. Workaround could be to test bulk-api-adapter end-2-end.

File: src/handlers/notification/index.js (hunk 16 above)
Comment (rmothilal) [reply src/handlers/notification/index.js]: This is not addressed

File: src/handlers/notification/index.js (hunk 17 above)
Comment (rmothilal) [reply src/handlers/notification/index.js]: This is not addressed

File: src/handlers/notification/index.js (hunk 18 above)
Comment (rmothilal) [reply src/handlers/notification/index.js]: this is not addressed

File: src/domain/bulkTransfer/transformer.js [hunk 19]
```diff
@@ -57,14 +57,14 @@ const transformHeaders = (headers, config) => {
   var normalizedHeaders = {}
 
   // check to see if FSPIOP-Destination header has been left out of the initial request. If so then add it.
-  if (!normalizedKeys.hasOwnProperty(ENUM.headers.FSPIOP.DESTINATION)) {
-    headers[ENUM.headers.FSPIOP.DESTINATION] = ''
+  if (!normalizedKeys.hasOwnProperty(HTTPENUM.headers.FSPIOP.DESTINATION)) {
```
Comment (rmothilal) [inline src/domain/bulkTransfer/transformer.js]: incorrect case supposed to be HTTPENUM.Headers

File: src/domain/bulkTransfer/transformer.js [hunk 20]
```diff
@@ -57,14 +57,14 @@ const transformHeaders = (headers, config) => {
...
+    headers[HTTPENUM.headers.FSPIOP.DESTINATION] = ''
```
Comment (rmothilal) [inline src/domain/bulkTransfer/transformer.js]: incorrect case supposed to be HTTPENUM.Headers

File: src/domain/bulkTransfer/transformer.js [hunk 21]
```diff
@@ -57,14 +57,14 @@ const transformHeaders = (headers, config) => {
...
   }
 
   for (var headerKey in headers) {
     var headerValue = headers[headerKey]
     switch (headerKey.toLowerCase()) {
-      case (ENUM.headers.GENERAL.DATE):
+      case (HTTPENUM.headers.GENERAL.DATE):
```
Comment (rmothilal) [inline src/domain/bulkTransfer/transformer.js]: incorrect case supposed to be HTTPENUM.Headers

File: src/domain/bulkTransfer/transformer.js [hunk 22]
```diff
@@ -80,21 +80,21 @@ const transformHeaders = (headers, config) => {
         }
         normalizedHeaders[headerKey] = tempDate
         break
-      case (ENUM.headers.GENERAL.CONTENT_LENGTH):
+      case (HTTPENUM.headers.GENERAL.CONTENT_LENGTH):
```
Comment (rmothilal) [inline src/domain/bulkTransfer/transformer.js]: incorrect case supposed to be HTTPENUM.Headers

File: src/domain/bulkTransfer/transformer.js [hunk 23]
```diff
@@ -80,21 +80,21 @@ const transformHeaders = (headers, config) => {
...
         // Do nothing here, do not map. This will be inserted correctly by the Hapi framework.
         break
-      case (ENUM.headers.FSPIOP.URI):
+      case (HTTPENUM.headers.FSPIOP.URI):
```
Comment (rmothilal) [inline src/domain/bulkTransfer/transformer.js]: incorrect case supposed to be HTTPENUM.Headers

File: src/domain/bulkTransfer/transformer.js [hunk 24]
```diff
@@ -80,21 +80,21 @@ const transformHeaders = (headers, config) => {
...
         // Check to see if we find a regex match the source header containing the switch name.
         // If so we include the uri otherwise we remove it.
 
-        if (headers[normalizedKeys[ENUM.headers.FSPIOP.SOURCE]].match(ENUM.headers.FSPIOP.SWITCH.regex) === null) {
+        if (headers[normalizedKeys[HTTPENUM.headers.FSPIOP.SOURCE]].match(HTTPENUM.headers.FSPIOP.SWITCH.regex) === null) {
```
Comment (rmothilal) [inline src/domain/bulkTransfer/transformer.js]: incorrect case supposed to be HTTPENUM.Headers

File: src/domain/bulkTransfer/transformer.js [hunk 25]
```diff
@@ -80,21 +80,21 @@ const transformHeaders = (headers, config) => {
...
           normalizedHeaders[headerKey] = headerValue
         }
         break
-      case (ENUM.headers.FSPIOP.HTTP_METHOD):
+      case (HTTPENUM.headers.FSPIOP.HTTP_METHOD):
```
Comment (rmothilal) [inline src/domain/bulkTransfer/transformer.js]: incorrect case supposed to be HTTPENUM.Headers

File: src/domain/bulkTransfer/transformer.js [hunk 26]
```diff
@@ -80,21 +80,21 @@ const transformHeaders = (headers, config) => {
...
         // Check to see if we find a regex match the source header containing the switch name.
         // If so we include the signature otherwise we remove it.
-        if (headers[normalizedKeys[ENUM.headers.FSPIOP.SOURCE]].match(ENUM.headers.FSPIOP.SWITCH.regex) === null) {
+        if (headers[normalizedKeys[HTTPENUM.headers.FSPIOP.SOURCE]].match(HTTPENUM.headers.FSPIOP.SWITCH.regex) === null) {
```
Comment (rmothilal) [inline src/domain/bulkTransfer/transformer.js]: incorrect case supposed to be HTTPENUM.Headers

File: src/domain/bulkTransfer/transformer.js [hunk 27]
```diff
@@ -112,27 +112,27 @@ const transformHeaders = (headers, config) => {
           }
         }
         break
-      case (ENUM.headers.FSPIOP.SOURCE):
+      case (HTTPENUM.headers.FSPIOP.SOURCE):
```
Comment (rmothilal) [inline src/domain/bulkTransfer/transformer.js]: incorrect case supposed to be HTTPENUM.Headers

File: src/domain/bulkTransfer/transformer.js [hunk 28]
```diff
@@ -112,27 +112,27 @@ const transformHeaders = (headers, config) => {
...
         normalizedHeaders[headerKey] = config.sourceFsp
         break
-      case (ENUM.headers.FSPIOP.DESTINATION):
+      case (HTTPENUM.headers.FSPIOP.DESTINATION):
```
Comment (rmothilal) [inline src/domain/bulkTransfer/transformer.js]: incorrect case supposed to be HTTPENUM.Headers

File: src/domain/bulkTransfer/transformer.js [hunk 29]
```diff
@@ -112,27 +112,27 @@ const transformHeaders = (headers, config) => {
...
         normalizedHeaders[headerKey] = config.destinationFsp
         break
       default:
         normalizedHeaders[headerKey] = headerValue
     }
   }
 
-  if (normalizedHeaders[normalizedKeys[ENUM.headers.FSPIOP.SOURCE]].match(ENUM.headers.FSPIOP.SWITCH.regex) !== null) {
+  if (normalizedHeaders[normalizedKeys[HTTPENUM.headers.FSPIOP.SOURCE]].match(HTTPENUM.headers.FSPIOP.SWITCH.regex) !== null) {
```
Comment (rmothilal) [inline src/domain/bulkTransfer/transformer.js]: incorrect case supposed to be HTTPENUM.Headers

File: src/domain/bulkTransfer/transformer.js [hunk 30]
```diff
@@ -112,27 +112,27 @@ const transformHeaders = (headers, config) => {
...
     // Check to see if we find a regex match the source header containing the switch name.
     // If so we remove the signature added by default.
-    delete normalizedHeaders[ENUM.headers.FSPIOP.SIGNATURE]
+    delete normalizedHeaders[HTTPENUM.headers.FSPIOP.SIGNATURE]
```
Comment (rmothilal) [inline src/domain/bulkTransfer/transformer.js]: incorrect case supposed to be HTTPENUM.Headers

File: src/domain/bulkTransfer/transformer.js [hunk 31]
```diff
@@ -112,27 +112,27 @@ const transformHeaders = (headers, config) => {
...
     // Aslo remove FSPIOP-URI and make FSPIOP-HTTP-Method ALL-CAPS #737
-    delete normalizedHeaders[ENUM.headers.FSPIOP.URI]
+    delete normalizedHeaders[HTTPENUM.headers.FSPIOP.URI]
```
Comment (rmothilal) [inline src/domain/bulkTransfer/transformer.js]: incorrect case supposed to be HTTPENUM.Headers

File: src/domain/bulkTransfer/transformer.js [hunk 32]
```diff
@@ -112,27 +112,27 @@ const transformHeaders = (headers, config) => {
...
   }
 
-  if (config && config.httpMethod !== ENUM.methods.FSPIOP_CALLBACK_URL_TRANSFER_POST) {
-    delete normalizedHeaders[ENUM.headers.GENERAL.ACCEPT]
+  if (config && config.httpMethod !== HTTPENUM.methods.FSPIOP_CALLBACK_URL_TRANSFER_POST) {
```
Comment (rmothilal) [inline src/domain/bulkTransfer/transformer.js]: incorrect case supposed to be HTTPENUM.Headers

File: src/domain/bulkTransfer/transformer.js [hunk 33]
```diff
@@ -112,27 +112,27 @@ const transformHeaders = (headers, config) => {
...
+    delete normalizedHeaders[HTTPENUM.headers.GENERAL.ACCEPT]
```
Comment (rmothilal) [inline src/domain/bulkTransfer/transformer.js]: incorrect case supposed to be HTTPENUM.Headers

Comment (rmothilal) [reply src/domain/bulkTransfer/transformer.js]: HTTPENUM.methods.FSPIOP_CALLBACK_URL_TRANSFER_POST needs to be 
HTTPENUM.RestMethods.POST

File: src/handlers/notification/index.js (hunk 16 above)
Comment (gibaros) [reply src/handlers/notification/index.js]: Addressed on latest commit

File: src/handlers/notification/index.js (hunk 17 above)
Comment (gibaros) [reply src/handlers/notification/index.js]: Addressed on latest commit

File: src/handlers/notification/index.js (hunk 18 above)
Comment (gibaros) [reply src/handlers/notification/index.js]: Addressed on latest commit

**Comment by gibaros** [comment]
PR ready for review, latest MASTER merged into branch, unit tests are now passing.

**Pull request merged**